src/screens/MessageScreen.tsx

```diff
@@ -1,37 +1,112 @@
-import React, { useCallback, useLayoutEffect, useMemo } from 'react';
-import { useHasNewMessagesFromUsers, useStyles, useUser } from '../hooks';
+import React, { useCallback, useLayoutEffect, useMemo, useState } from 'react';
+import { useStyles } from '../hooks';
 import { Avatar, Badge, Divider, List } from 'react-native-paper';
-import { TouchableOpacity } from 'react-native';
+import {
+  NativeScrollEvent,
+  NativeSyntheticEvent,
+  TouchableOpacity,
+} from 'react-native';
 import { ScrollView, View } from 'react-native';
 import { createStyles, useIcons } from '../components/BrandConfigProvider';
 import { HomeStackScreenProps } from '../navigators/types';
 import { t } from 'i18next';
-import { compact, orderBy } from 'lodash';
+import { chunk, compact, orderBy } from 'lodash';
 import { useLookupUsers } from '../hooks/Circles/usePrivatePosts';
 import { ActivityIndicatorView } from '../components';
 import { tID } from '../common';
+import { useUnreadMessages } from '../hooks/useUnreadMessages';
+import { useNotificationManager } from '../hooks/useNotificationManager';
+import { useFocusEffect } from '@react-navigation/native';
 
 export function MessageScreen({
   navigation,
   route,
 }: HomeStackScreenProps<'Home/Messages'>) {
   const { recipientsUserIds } = route.params;
-  const { styles } = useStyles(defaultStyles);
-  const { data, isLoading: userLoading } = useUser();
-
   useLayoutEffect(() => {
     navigation.setOptions({
       title: t('messages-title', 'My Messages'),
     });
   }, [navigation]);
 
+  const { styles } = useStyles(defaultStyles);
   const { User } = useIcons();
+  const [scrollIndex, setScrollIndex] = useState<number>(0);
+
+  // Get userIds that have sent new messages
+  const { unreadMessagesUserIds: unreadUserIds } = useUnreadMessages();
+
+  // Sort recipientsList by unread messages
+  const sortedList = useMemo(
+    () =>
+      orderBy(
+        recipientsUserIds,
+        (userId) => unreadUserIds?.includes(userId),
+        'desc',
+      ),
+    [unreadUserIds, recipientsUserIds],
+  );
+
+  // Break-down list of users into smaller chunks
+  const recipientsListChunked = useMemo(
+    () => chunk(sortedList, 10),
```

Comment:
Does 10 recipients constitute an entire screen? What happens if the device is large, will all the users still eventually get loaded?

Reply:
Scrolling down will load the next chunk of users but it is possible that on a very tall device that 10 items may not fill the entire height of the screen BUT unread messages will always be added to the top regardless. It could make for a strange UX is it does not fill the entire screen because they will have to scroll down to load the next set of 10 users. I will consider how to make this more dynamic this in a follow up PR.

Reply:
Sounds good. Specifically for the tall device scenario I would be worried that there might not be enough pages to fully load all the users.

Reply:
I am thinking of the scenario where they have a tall device and only 11 people to message. All 11 people are visible when the app loads. There is no way to scroll to the next page so number 11 never gets loaded.
I wonder if React Native has an interaction observer similar to the web.

Reply:
💡 Okay, I follow you now.